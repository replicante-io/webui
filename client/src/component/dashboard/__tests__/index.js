'use strict';
//@flow

import '@babel/polyfill';
import { Provider } from 'react-redux';

import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import Dashboard from '../index';


const mockStore = configureStore([]);


describe('Dashboard', () => {
  describe('index', () => {

    test('renders correctly', () => {
      let store = mockStore({
        datafetch: new Map(),
        dashboard: {
          clusters: []
        },
        events: {
          events: []
        }
      });
      const tree = renderer.create(
        <Provider store={store}>
          <Dashboard />
        </Provider>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

  });
});
