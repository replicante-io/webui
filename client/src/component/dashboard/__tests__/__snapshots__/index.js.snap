// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard index renders correctly 1`] = `
<div
  className="overview-container"
>
  <div
    className="row space-below"
  >
    <div
      className="col"
    >
      <div
        className="card"
      >
        <div
          className="card-header"
        >
          Largest clusters
          <button
            className="btn btn-outline-secondary fetch-button"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-rotate-right "
              data-icon="rotate-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M468.9 32.11c13.87 0 27.18 10.77 27.18 27.04v145.9c0 10.59-8.584 19.17-19.17 19.17h-145.7c-16.28 0-27.06-13.32-27.06-27.2c0-6.634 2.461-13.4 7.96-18.9l45.12-45.14c-28.22-23.14-63.85-36.64-101.3-36.64c-88.09 0-159.8 71.69-159.8 159.8S167.8 415.9 255.9 415.9c73.14 0 89.44-38.31 115.1-38.31c18.48 0 31.97 15.04 31.97 31.96c0 35.04-81.59 70.41-147 70.41c-123.4 0-223.9-100.5-223.9-223.9S132.6 32.44 256 32.44c54.6 0 106.2 20.39 146.4 55.26l47.6-47.63C455.5 34.57 462.3 32.11 468.9 32.11z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </button>
        </div>
        <div
          className="card-body text-center"
        >
          <p
            className="card-text large-icon"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-chart-area "
              data-icon="chart-area"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M64 400C64 408.8 71.16 416 80 416H480C497.7 416 512 430.3 512 448C512 465.7 497.7 480 480 480H80C35.82 480 0 444.2 0 400V64C0 46.33 14.33 32 32 32C49.67 32 64 46.33 64 64V400zM128 320V236C128 228.3 130.8 220.8 135.9 214.1L215.3 124.2C228.3 109.4 251.4 109.7 263.1 124.8L303.2 171.8C312.2 182.7 328.6 183.4 338.6 173.4L359.6 152.4C372.7 139.3 394.4 140.1 406.5 154.2L472.3 231C477.3 236.8 480 244.2 480 251.8V320C480 337.7 465.7 352 448 352H159.1C142.3 352 127.1 337.7 127.1 320L128 320z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </p>
          <p
            className="card-text"
          >
            No data available
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row space-below"
  >
    <div
      className="col"
    >
      <div
        className="card"
      >
        <div
          className="card-header"
        >
          Latest events
          <button
            className="btn btn-outline-secondary fetch-button"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-rotate-right "
              data-icon="rotate-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M468.9 32.11c13.87 0 27.18 10.77 27.18 27.04v145.9c0 10.59-8.584 19.17-19.17 19.17h-145.7c-16.28 0-27.06-13.32-27.06-27.2c0-6.634 2.461-13.4 7.96-18.9l45.12-45.14c-28.22-23.14-63.85-36.64-101.3-36.64c-88.09 0-159.8 71.69-159.8 159.8S167.8 415.9 255.9 415.9c73.14 0 89.44-38.31 115.1-38.31c18.48 0 31.97 15.04 31.97 31.96c0 35.04-81.59 70.41-147 70.41c-123.4 0-223.9-100.5-223.9-223.9S132.6 32.44 256 32.44c54.6 0 106.2 20.39 146.4 55.26l47.6-47.63C455.5 34.57 462.3 32.11 468.9 32.11z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </button>
        </div>
        <div
          className="card-body text-center"
        >
          <p
            className="card-text large-icon"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-chart-area "
              data-icon="chart-area"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M64 400C64 408.8 71.16 416 80 416H480C497.7 416 512 430.3 512 448C512 465.7 497.7 480 480 480H80C35.82 480 0 444.2 0 400V64C0 46.33 14.33 32 32 32C49.67 32 64 46.33 64 64V400zM128 320V236C128 228.3 130.8 220.8 135.9 214.1L215.3 124.2C228.3 109.4 251.4 109.7 263.1 124.8L303.2 171.8C312.2 182.7 328.6 183.4 338.6 173.4L359.6 152.4C372.7 139.3 394.4 140.1 406.5 154.2L472.3 231C477.3 236.8 480 244.2 480 251.8V320C480 337.7 465.7 352 448 352H159.1C142.3 352 127.1 337.7 127.1 320L128 320z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </p>
          <p
            className="card-text"
          >
            No data available
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
